src/librbd/ManagedLock.cc

```diff
@@ -481,14 +481,21 @@ void ManagedLock<I>::send_acquire_lock() {
 
   uint64_t watch_handle = m_watcher->get_watch_handle();
   if (watch_handle == 0) {
-    lderr(m_cct) << "watcher not registered - delaying request" << dendl;
-    m_state = STATE_WAITING_FOR_REGISTER;
+    if (m_watcher->is_blocklisted()) {
+      m_state = STATE_UNLOCKED;
+      lderr(m_cct) << "watcher is blocklisted - ending request" << dendl;
```

Comment:
This would be more correct IMO:
```suggestion
      lderr(m_cct) << "watcher not registered - client blocklisted" << dendl;
```